src/app/main.cpp

```diff
@@ -1538,7 +1553,10 @@ int main( int argc, char *argv[] )
     //replace backslashes with forward slashes
     pythonfile.replace( '\\', '/' );
 #endif
-    QgsPythonRunner::run( QStringLiteral( "with open('%1','r') as f: exec(f.read())" ).arg( pythonfile ) );
+    pythonfileArgs.prepend( pythonfile );
+    QgsPythonRunner::run( QStringLiteral( "sys.argv = ['%1']\n"
```

Comment:
I think this is not the best way to pass an array from c++ to python. Do someone know a better way?